### ISSUE TITLE
ISR does not work in App router nextjs: 13.5.3.

### ISSUE THREAD

**Issue by surething171** [vercel/next.js#56232]
### Link to the code that reproduces this issue

https://github.com/surething171/bug

### To Reproduce

1. Build the application `npm run build`
2. start the application `npm run start`
3. Open http://localhost:3000/ in any browser.
4. Refresh it couple of time also press `ctrl+shift+r` on chromium based browser.
5. Open http://localhost:3000/ in another incognito window. It is showing blank page at this stage. In microsoft edge I looked into network tab and for localhost request it is giving this response "Failed to load response data: No resource with given identifier found"
6. Same thing happen when I use fetch for some other backend routes.

### Current vs. Expected behavior

I expected the page to show updated information after refresh after 3 seconds, but it is showing same information. And on the incognito page it is not showing any information. It is only showing blank page. 

### Verify canary release

- [x] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #93-Ubuntu SMP Tue Sep 5 17:16:10 UTC 2023
Binaries:
  Node: 18.17.1
  npm: 9.6.7
  Yarn: 1.22.19
  pnpm: 8.8.0
Relevant Packages:
  next: 13.5.3
  eslint-config-next: 13.5.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Additional context

Everything was working fine when using pages router, am I missing something for the app router here? Please help.

**Comment by sleeporsmile** [comment]
same issue

**Comment by anthonyleeper** [comment]
Same issue. Fyi, this was working in 13.4.19.

**Comment by surething171** [comment]
@anthonyleeper, thanks, it is working in 13.4.19. I am reverting back to that version. If you find any other solution please let me know.

**Comment by Wunhyeon** [comment]
please fix, please~~~ 🥹

**Comment by Etheirystech** [comment]
The issue still persists in 14.0.0 as well.

**Comment by StefanoLazzaroni** [comment]
I can confirm, while building, next console doesn't recognize next revalidation as ISR, it computes it as Static + Json

**Comment by hallatore** [comment]
I tried this with `"next": "canary"` and refresh always gives me a new value after `3 seconds + 1 refresh` here.

**Comment by ShawnAndrews** [comment]
I can confirm this issue was fixed in v14.0.2 per code commit https://github.com/vercel/next.js/pull/57737. This issue affected versions v13.4.19 to v14.0.2 (exclusive) which was a period of 2-months. It is completely unacceptable for such a core feature like ISR to be broken and ignored while major versions continue to be released. This issue can be closed @surething171
